### PR TITLE
Added information about the app on the Home Page and tests

### LINKED ISSUE #1: FEATURE: Add information about the app on the Home Page
# User Story

* As a user 
* I can see what this application is about on the home page.
* So that the home page is more informative

### Discussion

As you log into the app you will see "greetings @cgaucho" and a lot of blank space. 

So the main task here is to fill that with some information about the app. Here are some suggestions:

```
This app is intended as a replacement for the <a href="https://ucsb-cs-github-linker.herokuapp.com">ucsb-cs-github-linker</a> app used in many courses at UCSB, as well as some courses at other universities.   
```

### Acceptance Criteria

- [ ] There should be some actual content on the whitespace except for plain white space.

### Implementation Todos

Frontend:
- [ ] Make changes in HomePage.js file.

Testing:
- [ ] Front end tests pass and there is adequate coverage

### PR DESCRIPTION
**Pull request by 30912hyl**
Added a line explaining purpose of app in Homepage

Closes #1

### CODE REVIEW

**Review by pconrad — CHANGES_REQUESTED** [review]
I know it's not your fault, but this Home Page implementation is a mess, and now that you've touched it, it's your responsibility to clean it up, I'm afraid.

There shouldn't be two huge branches of code for a page where someone's logged in vs. a version where they aren't.  Literally, the only difference is this:

```
{greeting}, {currentUser.root.user.githubLogin}
```
vs. 
```
{greeting}, cgaucho
```

No!  We should just make the greeting conditional on being logged in.

One render block,  containing:

```
{ (!currentUser || !currentUser.loggedIn) && (  <>{greeting}, {currentUser.root.user.githubLogin}</> )  }
```

in place of where you now see: 

```
{greeting}, {currentUser.root.user.githubLogin}
```

**Review by pconrad — CHANGES_REQUESTED** [review]
Why are there changes to the SchoolEditPage in this branch?

Each PR should deal with one and only one concern.

Always start from a fresh branch of main when starting a new story UNLESS the new story DEPENDS on code in the older story.  These two are completely independent of one another and so should not be in the same branch.

If you need help on disentangling the branches, let us know.  We can help.

**Review by sanilkatula — CHANGES_REQUESTED** [review]
`package edu.ucsb.cs156.organic.interceptors;`


Keep all the interceptors with small case "i" only!

**Review by pconrad — CHANGES_REQUESTED** [review]
Please read: https://ucsb-cs156.github.io/topics/pull_requests/package_lock_json.html and then do what's needed to get package-lock.json restored to the original values.

**Review by pconrad — CHANGES_REQUESTED** [review]
The home page needs a major refactor.  And it isn't that difficult to accomplish, but maybe I didn't provide enough explanation last time. Let me try again.

It isn't your fault that you are making a change to some bad code.   But that will happen a LOT later in life (at jobs or internships).  And by not cleaning it up, you are *making the problem worse*. 

Instead, you can be the hero that makes the code better. That's how you want to be known on your team, in this course, and in later life.  It's a skill worth learning.

Right now, the page logic has two parts:

* a part that is done when this condition is true: `(!currentUser || !currentUser.loggedIn)` * a part that is done when that condition is false.

And the *entire* logic is repeated when there's really only the tinest difference between the two branches.  

That's ... really bad code.  It's bad because it's hard to maintain. For example, you had to add the new home page information in TWO different places.  The previous developer created a trap that doubles your work.   By taking the bait, you're making the problem worse: now the next developer that needs to make a change has even MORE code that has do be maintained separately.

Please don't do that... not in this course, but especially, not after you graduate and are representing UCSB out in the world.


A better way to structure this is to put the condition where it will cause the least amount of repeated code.  Here's one approach:

Instead of this, which is how the code is structured now:

```
if (condition) return (
     details one
     details two
     details three
     GREETINGS CGAUCHO
     details four 
} else {
  details one
     details two
     details three
     GREETINGS {actual name}
     details four 
}
```

Instead: 

const nameToDisplay = condition ? "cgaucho" : {real name}
return (

  details one
     details two
     details three
     greetings { name_to_display }
     details four 
}
```

Do you see how this makes the code so much easier to maintain?

Also: please remove `cgaucho`.  Instead of displaying `cgaucho`, just show no greeting at all, like this:

const greeting = condition ? "" : "greeting " + {real name};

Then, use {greeting} inside the component.

**Comment by 30912hyl** [comment]
Hi, thanks for all the help; I tried my best to implement and improve the code based on the proposed changes. Regarding @sanilkatula comment on using lowercase "i" for interceptors, I assume that the "Interceptors" folder is incorrectly named and should be changed to "interceptors" matching the lowercasing of the other folders, such that the import statements don't result in errors. I'm not too sure how to add this rename however, as I believe git tracks changes to a file rather than folder names? In this PR, I decided on just removing the changed "import Interceptors" files instead. Let me know if there's something else I should have done instead.

**Review by Xalex217 — DISMISSED** [review]
lgtm

**Review by pconrad — CHANGES_REQUESTED** [review]
Commented out code needs to be removed.
https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

**Comment by pconrad** [comment]
Closing per converation with Aaron on the slack channel.. he's going to replace with a different PR

**Comment by 30912hyl** [comment]
Replaced with [PR #52](https://github.com/ucsb-cs156-s24/proj-organic-s24-5pm-3/pull/52)
